File: wardrobe-services/wardrobe-service/src/main/java/com/example/dto/response/ItemResponse.java
package com.example.dto.response;


import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.Instant;
import java.util.UUID;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class ItemResponse {

    private UUID id;
    private String name;
    private String type;
    private String color;
    private String season;
    private String brand;
    private String occasion;
    private String imageUrl;
    private Instant createdAt;

}

File: fashion_outfit_suggestions_app/lib/core/models/item_response.dart
class ItemResponse {
  final String id;
  final String name;
  final String type;
  final String color;
  final String? season;
  final String? brand;
  final String? occasion;
  final String? imageUrl;
  final String? tags;

  ItemResponse({
    required this.id,
    required this.name,
    required this.type,
    required this.color,
    this.season,
    this.brand,
    this.occasion,
    this.imageUrl,
    this.tags,
  });

  factory ItemResponse.fromJson(Map<String, dynamic> json) {
    return ItemResponse(
      id: json['id']?.toString() ?? '',
      name: json['name'] as String? ?? '',
      type: json['type'] as String? ?? '',
      color: json['color'] as String? ?? '',
      season: json['season'] as String?,
      brand: json['brand'] as String?,
      occasion: json['occasion'] as String?,
      imageUrl: json['imageUrl'] as String?,
      tags: json['tags'] as String?,
    );
  }
}
